### DISCUSSION TITLE
Storing objects created by entry functions offchain

### DISCUSSION THREAD

**Discussion by ajaythxkur** [aptos-labs/aptos-developer-discussions#401] [Questions] [answered]
In entry functions we create objects and entry functions don't return anythting. How is it possible to store object address offchain in db??

**Comment by kshitijc1506** [comment]
You can emit an event with the object address.

**Reply by ajaythxkur** [reply]
aptosClient.getModuleEventsByEventType
is there any way to subscribe to the events to get real time data??

**Reply by ajaythxkur** [reply]
or is it possible to get a particular event from tx hash??

**Reply by kshitijc1506** [reply]
You could probably just use the [indexer API](https://cloud.hasura.io/public/graphiql?endpoint=https://api.mainnet.aptoslabs.com/v1/graphql) or create a [custom processor](https://aptos.dev/en/build/indexer/custom-processors) if needed. I don't believe you can query events from txn hash today, but there are multiple ways to extract events as you can see in the hasura playground. You can do it for a particular account address, for instance.

**Reply by ajaythxkur** [reply, marked as answer]
@kshitjee i can do one more thing. 
Get the tx hash, filter out the event and get the data
this will work.
is it correct method??

something like this
```
const transaction = await aptos.getTransactionByHash({ transactionHash: response.hash });
                if(transaction.type === "user_transaction"){
                    const event = transaction.events.find((event) => event.type === `${ABI_ADDRESS}::main::MyEvent`);
                    if(event){
                        
                    }
                }
```

i know that the type of txn is user txn and i know this will emit that specific event for my app. so this might work

**Reply by kshitijc1506** [reply]
Yeah that LGTM

**Reply by ajaythxkur** [reply]
Alright thanks @kshitjee
